.test-defs/cmd/shoot-maintenance/main.go

```diff
@@ -57,11 +60,98 @@ var (
 	testMachineImage = gardenv1beta1.ShootMachineImage{
 		Version: testMachineImageVersion,
 	}
+
+	shootMaintenanceTest          *framework.ShootMaintenanceTest
+	gardenerSchedulerReplicaCount *int32
+
+	ctxSetup, cancelCtxSetup     = context.WithTimeout(context.Background(), time.Minute*2)
```

Comment:
When we define all contexts at the beginning isn't the timer of all contexts starting at the same time during startup?
This would mean that the timer of `ctxRestore` will timeout after 2 minutes after the program started whereas `ctxMain` will timeout after 5 minutes. What we want is that the timer of `ctxRestore` starts when the function `restoreEnvironment` gets called so that we only measure the execution of this one function.

Therefore I suggest to define the contexts in the function itself and only define the timeout constants in the init section.

Reply:
Good point